src/main/java/teammates/ui/template/CourseEditInstructorPanel.java

```diff
@@ -56,48 +56,90 @@ public CourseEditInstructorPanel(int instructorToShowIndex, int instructorIndex,
     private List<CourseEditSectionRow> createSectionRows(int instructorIndex, List<String> sectionNames,
                                    List<String> feedbackNames) {
         firstBlankSectionRowIndex = sectionNames.size();
-        List<CourseEditSectionRow> rows = new ArrayList<CourseEditSectionRow>();
         Map<Integer, String> specialSectionNames = new TreeMap<Integer, String>();
         Map<Integer, String> nonSpecialSectionNames = new TreeMap<Integer, String>();
         
-        int sectionIndex = -1;
+        distinguishSpecialAndNonSpecialSections(sectionNames, specialSectionNames, nonSpecialSectionNames);
+
+        List<CourseEditSectionRow> rows =
+                createSpecialAndNonSpecialSectionRowsInOrder(instructorIndex, sectionNames, feedbackNames,
+                                                             specialSectionNames, nonSpecialSectionNames);
+        return rows;
+    }
+    
+    /**
+     * Splits the list of section names into two mappings of Integer => String
```

Comment:
I feel that wording this as `two Integer to String mappings` instead of `two mappings of Integer => String` is a little clearer